flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/calcite/PreValidateReWriter.scala

```diff
@@ -391,8 +392,19 @@ object PreValidateReWriter {
       == desiredType)) {
       node
     } else {
-      SqlStdOperatorTable.CAST.createCall(SqlParserPos.ZERO,
-        node, SqlTypeUtil.convertTypeToSpec(desiredType))
+      // See FLINK-26460 for more details
+      val sqlDataTypeSpec =
+        if (currentType.getSqlTypeName == SqlTypeName.NULL
+          && desiredType.getSqlTypeName == SqlTypeName.MAP) {
+          new SqlDataTypeSpec(
+            new SqlMapTypeNameSpec(SqlTypeUtil.convertTypeToSpec(desiredType.getKeyType),
```

Comment:
we should consider whether the key type and value type are not null, such as: MAP<STRING NOT NULL, BIGINT>, MAP<STRING NOT NULL, BIGINT NOT NULL>

Reply:
Thanks for remind, I has fix it.